clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -705,8 +731,15 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                 // Verify the number of read/write records matches the last
                 // operand. Skip D-filtered traces which don't have every load or
                 // store records.
-                report_if_false(shard, shard->expected_read_records_ == 0,
+                report_if_false(shard,
+                                shard->expected_read_records_ == 0 ||
+                                    // Some prefetches did not have any corresponding
+                                    // memory access in system call trace templates
+                                    // collected on QEMU.
```

Comment:
Please elaborate: do we consider that a bug?  QEMU treats prefetch loads as not normal loads and doesn't put them into traces?  Put an issue # where this is discussed?  I see no mention in #6495.

Reply:
I figured it is simply a case of QEMU not honoring prefetch instrs and just not issuing a real memory access.